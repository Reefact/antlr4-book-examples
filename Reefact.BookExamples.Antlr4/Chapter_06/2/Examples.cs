#region Usings declarations

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._2 {

    [UseReporter(typeof(BeyondCompareReporter))]
    public class Examples {

        [Fact]
        public void t_json_tokensString() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("t.json", 6, 2);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void t_json_mermaid_style_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("t.json", 6, 2);
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

        [Fact]
        public void simple_array_tokens() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("[1,\"\\u0049\",1.3e9]");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        public void simple_array_tree() {
            // Setup
            AntlrInputStream inputStream = AntlrInputStreamReader.Read("[1,\"\\u0049\",1.3e9]");
            GRun             grun        = GRun.Read(inputStream);
            // Exercise
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(lispStyleTree);
        }

    }

}