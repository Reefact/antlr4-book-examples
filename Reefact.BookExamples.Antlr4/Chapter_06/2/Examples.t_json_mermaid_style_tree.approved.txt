graph TD
	1["json"] --> 2["object"]
	2 --> 3["{"]
	2 --> 4["pair"]
	4 --> 5["#0034;antlr.org#0034;"]
	4 --> 6[":"]
	4 --> 7["value"]
	7 --> 8["object"]
	8 --> 9["{"]
	8 --> 10["pair"]
	10 --> 11["#0034;owners#0034;"]
	10 --> 12[":"]
	10 --> 13["value"]
	13 --> 14["array"]
	14 --> 15["["]
	14 --> 16["]"]
	8 --> 17[","]
	8 --> 18["pair"]
	18 --> 19["#0034;live#0034;"]
	18 --> 20[":"]
	18 --> 21["value"]
	21 --> 22["true"]
	8 --> 23[","]
	8 --> 24["pair"]
	24 --> 25["#0034;speed#0034;"]
	24 --> 26[":"]
	24 --> 27["value"]
	27 --> 28["1e100"]
	8 --> 29[","]
	8 --> 30["pair"]
	30 --> 31["#0034;menus#0034;"]
	30 --> 32[":"]
	30 --> 33["value"]
	33 --> 34["array"]
	34 --> 35["["]
	34 --> 36["value"]
	36 --> 37["#0034;File#0034;"]
	34 --> 38[","]
	34 --> 39["value"]
	39 --> 40["#0034;Help#0092;nMenu#0034;"]
	34 --> 41["]"]
	8 --> 42["}"]
	2 --> 43["}"]

classDef default fill:#fff,stroke:#000,stroke-width:0.25px;
classDef error color:#fff,fill:#FF0000,stroke:#000,stroke-width:0.25px;